coin3d 4.0.0 no longer builds on nixos-unstable

**Describe the bug**
When installing the [`coin3d` package](https://github.com/NixOS/nixpkgs/blob/83cbad92d73216bb0d9187c56cce0b91f9121d5a/pkgs/development/libraries/coin3d/default.nix), the build fails with error:
```
<PRIOR OUTPUT OMITTED>
-- Performing Test HAVE_HASH_QUOTING - Success
-- Performing Test DISABLE_WARNING_TAUTOLOGCOMPARE
-- Performing Test DISABLE_WARNING_TAUTOLOGCOMPARE - Success
-- Configuring done
CMake Error: install(EXPORT "Coin" ...) includes target "Coin" more than once in the export set.
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTING
    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_POLICY_DEFAULT_CMP0025


CMake Generate step failed.  Build files cannot be regenerated correctly.
```

**To Reproduce**
Steps to reproduce the behavior:
1. Use `nixos-unstable` channel
2. `nix-shell -p pkgs.coin3d`, or any other method of build/installation

**Expected behavior**
`coin3d` can be installed explicitly or as a dependency.

**Additional context**
I encountered this issue while attempting to install `freecad`.  Curiously, I had installed `freecad` last week with no issues, and neither `coin3d` nor `freecad` have been modified since late September and October, respectively.

**Notify maintainers**
@gebner @viric 

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.9.12, NixOS, 21.03pre257339.83cbad92d73 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.9`
 - channels(leaf): `""`
 - channels(root): `"nixos-21.03pre257339.83cbad92d73, nixos-hardware"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/applications/graphics/freecad/default.nix
+++ b/pkgs/applications/graphics/freecad/default.nix
@@ -9,13 +9,13 @@ let
   pythonPackages = python3Packages;
 in mkDerivation rec {
   pname = "freecad-unstable";
-  version = "2020-10-17";
+  version = "2020-12-08";
 
   src = fetchFromGitHub {
     owner = "FreeCAD";
     repo = "FreeCAD";
-    rev = "f3bdaaa55a6c03b297924c40819d23e4603fa55b";
-    sha256 = "1q1iy4i9k65v8z7h8a6r4bf5ycn124jp26xwp0xwbar4gnkx2jiq";
+    rev = "daea30341ea2d5eaf2bfb65614128a5fa2abc8b7";
+    sha256 = "1fza64lygqq35v7kzgqmiq5dvl5rpgkhlzv06f9dszdz44hznina";
   };
 
   nativeBuildInputs = [

--- a/pkgs/development/libraries/coin3d/default.nix
+++ b/pkgs/development/libraries/coin3d/default.nix
@@ -1,14 +1,15 @@
-{ fetchFromGitHub, stdenv, boost, cmake, libGL, libGLU }:
+{ fetchFromGitHub, stdenv, boost, cmake, libX11, libGL, libGLU }:
 
 stdenv.mkDerivation rec {
   pname = "coin";
-  version = "4.0.0";
+  version = "2020-12-07-unstable";
 
   src = fetchFromGitHub {
     owner = "coin3d";
     repo = "coin";
-    rev = "Coin-${version}";
-    sha256 = "1ayg0hl8wanhadahm5xbghghxw1qjwqbrs3dl3ngnff027hsyf8p";
+    # rev = "Coin-${version}";
+    rev = "d5539998aff272b349590fe74d068659682ecd0d";
+    sha256 = "11jaz8p9nn8jpd6dsgwgkldwr7z829gyf64g014qyyh8l6p7jzzd";
   };
 
   postPatch = ''
@@ -16,7 +17,7 @@ stdenv.mkDerivation rec {
   '';
 
   nativeBuildInputs = [ cmake ];
-  buildInputs = [ boost libGL libGLU ];
+  buildInputs = [ boost libX11 libGL libGLU ];
 
   meta = with stdenv.lib; {
     homepage = "https://github.com/coin3d/coin";

--- a/pkgs/development/libraries/soqt/default.nix
+++ b/pkgs/development/libraries/soqt/default.nix
@@ -1,12 +1,16 @@
-{ fetchurl, stdenv, coin3d, qtbase, cmake, pkgconfig }:
+{ fetchFromGitHub, stdenv, coin3d, qtbase, cmake, pkgconfig }:
 
 stdenv.mkDerivation rec {
   pname = "soqt";
-  version = "1.6.0";
+  version = "2020-12-05-unstable";
 
-  src = fetchurl {
-    url = "https://github.com/coin3d/soqt/releases/download/SoQt-${version}/soqt-${version}-src.tar.gz";
-    sha256 = "07qfljy286vb7y1p93205zn9sp1lpn0rcrqm5010gj87kzsmllwz";
+  src = fetchFromGitHub {
+    owner = "coin3d";
+    repo = "soqt";
+    # rev = "SoQt-${version}";
+    rev = "fb8f655632bb9c9c60e0ff9fa69a5ba22d3ff99d";
+    sha256 = "sha256-YoBq8P3Tag2Sepqxf/qIcJDBhH/gladBmDUj78aacZs=";
+    fetchSubmodules = true;
   };
 
   buildInputs = [ coin3d qtbase ];

